packages/components/src/SubHeaderBar/SubHeaderBar.component.js

```diff
@@ -7,85 +8,130 @@ import { DEFAULT_I18N, getDefaultTranslate } from '../translate';
-					icon="talend-arrow-left"
-					bsStyle="link"
-					className={classNames(theme['tc-subheader-back-button'], 'tc-subheader-back-button')}
-					hideLabel
-				/>
-			),
-		},
-		{
-			injectedComponent: <InputTitleSubHeader t={t} {...rest} />,
-		},
-	];
+function CustomInject({ getComponent, left, right, center, nowrap, ...props }) {
+	if (nowrap) {
+		return <Inject getComponent={getComponent} {...props} />;
+	}
+	return (
+		<SubHeaderBarActions left={left} right={right} center={center}>
+			<Inject getComponent={getComponent} {...props} />
+		</SubHeaderBarActions>
+	);
+}
+CustomInject.propTypes = {
+	nowrap: PropTypes.bool,
+	right: PropTypes.bool,
+	center: PropTypes.bool,
+	left: PropTypes.bool,
+	getComponent: PropTypes.func,
+};
+
+function SubHeaderBar({
+	t,
+	onGoBack,
+	components,
+	getComponent,
+	className,
+	children,
+	left,
+	center,
+	right,
+	...rest
+}) {
+	const injected = Inject.all(getComponent, components, CustomInject);
+	const Components = Inject.getAll(getComponent, { Action, ActionBar });
+	const hasRight = (
+		Array.isArray(right) ||
+		has(components, 'before-right') ||
+		has(components, 'after-right')
+	);
 	return (
 		<header className={classNames(theme['tc-subheader'], 'tc-subheader', className)}>
-			<ActionBar className={classNames(theme['tc-subheader-navbar'], 'tc-subheader-navbar')}>
-				<SubHeaderBarActions
-					components={defaultComponentsLeft}
-					className={theme['tc-subheader-navbar-left']}
-					left
-					center={false}
-					right={false}
-				/>
-				{Array.isArray(componentsCenter) && (
-					<SubHeaderBarActions
-						components={componentsCenter}
-						className={classNames([theme['tc-subheader-navbar-center']], 'subheader-center', {
-							[`${theme['tc-subheader-navbar-center-no-margin-right']}`]: componentsRight,
-							'tc-subheader-navbar-center-no-margin-right': componentsRight,
-						})}
-						left={false}
-						center
-						right={false}
-					/>
+			{injected('before-actionbar')}
+			<Components.ActionBar className={classNames(theme['tc-subheader-navbar'], 'tc-subheader-navbar')}>
+				{injected('before-left')}
+				<SubHeaderBarActions left>
+					{injected('before-back')}
+					{onGoBack && (
+						<Components.Action
+							id="backArrow"
+							onClick={onGoBack}
+							label={t('BACK_ARROW_TOOLTIP', { defaultValue: 'Go Back' })}
+							icon="talend-arrow-left"
+							bsStyle="link"
+							className={classNames(theme['tc-subheader-back-button'], 'tc-subheader-back-button')}
+							hideLabel
+						/>
+					)}
+					{injected('before-title')}
+					<InputTitleSubHeader t={t} {...rest} />
+					{injected('after-title')}
+				</SubHeaderBarActions>
+				{Array.isArray(left) && (
+					<SubHeaderBarActions left>
+						{left.map((item, index) => (
+							<Components.Action key={index} {...item} />
+						))}
+					</SubHeaderBarActions>
 				)}
-				{Array.isArray(componentsRight) && (
+				{children}
```

Comment:
Why do we need a `children` if we have the injection mechanism?

Reply:
guess it is in case you don't need/use it

Reply:
We can remove it for sure, I'm not anymore close to this, but if you are I'm happy with that !